tripy/tripy/frontend/module/sequential.py

```diff
@@ -0,0 +1,284 @@
+    added in the order they are passed, and each is called sequentially during the forward pass.
+    """
+    modules: Union[Module, Dict[str, Module]]
+    r"""The modules to include in the sequence"""
+
+    def __init__(self, *modules: Union[Module, Dict[str, Module]]) -> None:
+        r"""
+        Args:
+            *modules (Union[Module, Dict[str, Module]]): The modules to include in the sequence.
+                Can be passed as individual positional arguments or as a single dictionary of named modules.
+
+        .. code-block:: python
+            :linenos:
+            :caption: Example
+
+            import tripy as tp
+
+            # Sequential with layers passed as arguments
+            model = tp.Sequential(tp.Linear(1, 3), tp.Linear(3, 2))
+
+            # Sequential with named dictionary of layers
+            model_named = tp.Sequential({'layer1': tp.Linear(1, 3), 'layer2': tp.Linear(3, 2)})
+
+            # Forward pass
+            x = tp.Tensor([1.0])
+            output = model(x)
+        """
+        super().__init__()
+        self._modules = {}
+
+        if len(modules) == 1 and isinstance(modules[0], dict):
+            for name, module in modules[0].items():
+                self.add_module(name, module)
+        else:
+            for idx, module in enumerate(modules):
+                self.add_module(str(idx), module)
+
+    def __call__(self, x: "tripy.Tensor") -> "tripy.Tensor":
+        r"""
+        Defines the forward pass by applying each module in the container sequentially to input `x`.
+
+        Args:
+            x (Tensor): The input tensor to pass through the sequence of modules.
+
+        Returns:
+            Tensor: The output tensor after passing through each module in sequence.
+        """
+        for module in self._modules.values():
+            x = module(x)
+        return x
+    
+    def __len__(self) -> int:
+        r"""
+        Returns the total number of modules in the sequence.
+
+        Returns:
+            int: The number of modules in the sequence.
+        
+        .. code-block:: python
+            :linenos:
+            :caption: Example
+
+            model = tp.Sequential(tp.Linear(1, 64), tp.Linear(64, 128))
+            assert len(model) == 2
+        """
+        return len(self._modules)
+
+
+    def __iter__(self) -> Iterator[Module]:
+        r"""
+        Returns an iterator over the modules in the sequence.
+
+        Returns:
+            Iterator[Module]: An iterator over the modules.
+        
+        .. code-block:: python
+            :linenos:
+            :caption: Example
+
+            model = tp.Sequential(tp.Linear(1, 3), tp.Linear(3, 2))
+            for layer in model:
+                print(layer)
+        """
+        return iter(self._modules.values())
+
+    def __getitem__(self, idx: Union[int, str]) -> Module:
+        r"""
+        Accesses a module by index (int) or name (str).
```

Comment:
Is `int` here valid syntax? I think the interpreter would reject something like `module.0`.

Reply:
I think it would be fine. `__getitem__` is array-style access.

Reply:
Ah my mistake, I misread it as `__getattr__`.